From 6555d3f587e485afb1676948d7a5868afac3aee8 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Sun, 21 Jun 2020 19:48:54 -0700
Subject: [PATCH] move nixio__bin2hex to header and make static

This fixes compilation under GCC10.

There's no reason for this to be in a C file anyway.

Also fixed a -Wmissing-prototypes warning with nixio__addr_prefix.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 src/address.c | 2 +-
 src/binary.c  | 4 ----
 src/nixio.h   | 4 +++-
 3 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/address.c b/src/address.c
index 0bf53caf82..6ed3e5c0e9 100644
--- a/src/address.c
+++ b/src/address.c
@@ -153,7 +153,7 @@ int nixio__addr_write(nixio_addr *addr, struct sockaddr *saddr) {
 /**
  * netmask to prefix helper
  */
-int nixio__addr_prefix(struct sockaddr *saddr) {
+static int nixio__addr_prefix(struct sockaddr *saddr) {
 	int prefix = 0;
 	size_t len;
 	uint8_t *addr;
diff --git a/src/binary.c b/src/binary.c
index da794536ce..56f6d1e2f6 100644
--- a/src/binary.c
+++ b/src/binary.c
@@ -19,10 +19,6 @@
 #include "nixio.h"
 #include <stdlib.h>
 
-const char nixio__bin2hex[16] = {
-'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
-};
-
 static unsigned char nixio__b64encode_tbl[] =
 	"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
 
diff --git a/src/nixio.h b/src/nixio.h
index 5eb84d11ae..25874eb04f 100644
--- a/src/nixio.h
+++ b/src/nixio.h
@@ -107,7 +107,9 @@ int nixio__mode_write(int mode, char *modestr);
 
 int nixio__push_stat(lua_State *L, nixio_stat_t *buf);
 
-const char nixio__bin2hex[16];
+static const char nixio__bin2hex[16] = {
+'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
+};
 
 /* Module functions */
 void nixio_open_file(lua_State *L);
